### PR TITLE
update 'gen_summary_lt' to 'summarize_lt'

### LINKED ISSUE #67: FEATURE: Align `gen_summary_lt` with `demUtils::summarize_dt`
**Describe the solution you'd like**
`gen_summary_lt` should have a similar interface to `demUtils::summarize_dt` and have similar functionality.

Also recommend switching name to `summarize_lt` since we have been reserving `gen_` for functions that modify the input `dt` in place.

### PR DESCRIPTION
**Pull request by chacalle**
## Describe changes

Update 'gen_summary_lt' to 'summarize_lt' so that it has a similar interface as `summarize_dt`.

This would likely require changes in `model_lifetable` & `model_population`.

## What issues are related

Fixes #67

### Packages Repositories

* [X] Have you read the [contributing guidelines](https://github.com/ihmeuw-demographics/packageTemplate/wiki#guide-to-r-package-development) for `ihmeuw-demographics` R packages?
* [X] Have you successfully run `devtools::check()` locally?
* [X] Have you updated or added function (and vignette if applicable) documentation? Did you update the 'man' and 'NAMESPACE' files with `devtools::document()`?
* [X] Have you added in tests for the changes included in the PR?
* [X] Do the changes follow the `ihmeuw-demographics` [code style](https://github.com/ihmeuw-demographics/packageTemplate/wiki/Code-style-guide)?
* [ ] Do the changes need to be immediately included in a new build of [`docker-base`](https://github.com/ihmeuw-demographics/docker-base) or [`docker-internal`](https://github.com/ihmeuw-demographics/docker-internal)? If so follow directions in those repositories to rebuild and redeploy the images.
* [X] Do the changes require updates to other repositories which use this package? If yes, make the necessary updates in those repos, and consider integration tests for those repositories.

### CODE REVIEW

File: R/summarize_lt.R [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+#' )
+#'
+#' @export
+summarize_lt <- function(dt,
+                         id_cols,
+                         summarize_cols,
+                         value_cols,
+                         summary_fun = c("mean"),
+                         probs = c(0.025, 0.975),
+                         recalculate_stats = c("mean"),
```
Comment (chacalle) [inline R/summarize_lt.R:79]: Better name for this argument?

File: R/summarize_lt.R [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+#'   Value columns (life table parameters) that summary statistics should be
+#'   calculated for. Valid choices are 'mx', 'qx', 'ax,', 'dx', 'px', 'lx',
+#'   'dx', 'Tx', 'nLx', & 'ex'. Must include at least 2 of 'mx', 'qx', or 'ax'.
+#' @param recalculate_stats \[`character()`\]\cr
+#'   The summary statistic to recalculate all life table parameters
+#'   corresponding to that statistic so that they are consistent with one
+#'   another. Uses `demCore::lifetable` to recalculate all life table parameters
+#'   and keeps the life table parameters specified in `value_cols`. Default is
+#'   'mean'.
+#' @param format_long \[`logical(1)`\]\cr
```
Comment (chacalle) [inline R/summarize_lt.R:22]: Now thinking it may be helpful to add this to `demCore::lifetable` too for convenience

File: R/summarize_lt.R [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    summary_fun = summary_fun,
+    probs = probs
+  )
+  original_summary_cols <- names(summary_dt)
+  original_summary_keys <- key(summary_dt)
+
+  # recalculate mean --------------------------------------------------------
+
+  keep_params <- c("mx", "qx", "ax")
+  recalculate_params <- setdiff(value_cols, keep_params)
```
Comment (krpaulson) [inline R/summarize_lt.R:128]: But what if mx, ax, and qx don't all agree? Keep mx and ax and recalculate qx?

**Review by krpaulson — DISMISSED** [review]

File: R/summarize_lt.R [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+#'
+#' @details
+#' One example use case for `summarize_lt` is when we have multiple draws
+#' (independent simulations) of life tables to propagate uncertainty. Each of
+#' the independent draws of life tables may have all life table parameters. The
+#' mean, 2.5th and 97.5 percentiles across all draws for all life table
+#' parameters can be calculated. But the mean 'mx', 'qx', 'ax' parameters would
+#' be inconsistent with the mean 'ex' parameter for example. This is when
+#' specifying `recalculate_stats = 'mean'` would recalculate the mean 'ex'
+#' parameter using the mean 'mx', 'qx', and/or 'ax' as inputs to the
```
[the remark this answers is not in the document]
Comment (chacalle) [reply R/summarize_lt.R:43]: Went ahead and added `recalculate_starting_params = c("mx", "ax")`. 

Noticed that in `demCore::lifetable` we always recalculate `qx`. https://github.com/ihmeuw-demographics/demCore/blob/master/R/lifetable.R#L29-L31 At first I was thinking we need a similar argument there  but now I wonder if instead that should be a validation that the recalculated `qx` is consistent with the other two parameters.

File: R/summarize_lt.R (hunk 2 above)
Comment (chacalle) [reply R/summarize_lt.R:22]: Going to create a separate issue/PR for this to keep this PR short https://github.com/ihmeuw-demographics/demCore/issues/73

**Pull request merged**